DirectProgramming/DPC++FPGA/ReferenceDesigns/cholesky/src/cholesky.hpp

```diff
@@ -0,0 +1,158 @@
+  // and write their content to the AMatrixPipe pipe.
+  auto ddr_read_event = q.submit([&](sycl::handler &h) {
+    h.single_task<CholeskyDDRToLocalMem>([=]() [[intel::kernel_args_restrict]] {
+      MatrixReadFromDDRToPipe<TT, rows, columns, kNumElementsPerDDRBurst,
+                              AMatrixPipe>(a_device, matrix_count, repetitions);
+    });
+  });
+
+  // Read the A matrix from the AMatrixPipe pipe and compute the Cholesky
+  // decomposition. Write the L output matrix to the LMatrixPipe pipe.
+  q.single_task<Cholesky>(
+      fpga_linalg::StreamingCholesky<T, is_complex, rows, raw_latency,
+                        kNumElementsPerDDRBurst, AMatrixPipe, LMatrixPipe>());
+
+  auto ddr_write_event =
+  q.single_task<CholeskyLocalMemToDDRL>([=]() [[intel::kernel_args_restrict]] {
+    // Read the L matrix from the LMatrixPipe pipe and copy it to the FPGA DDR
+    // Number of DDR bursts of kNumElementsPerDDRBurst required to write
+    // one vector
+    constexpr bool kIncompleteBurst =
+       kLMatrixSize % kNumElementsPerDDRBurst != 0;
+    constexpr int kExtraIteration = kIncompleteBurst ? 1 : 0;
+    constexpr int kLoopIter =
+       (kLMatrixSize / kNumElementsPerDDRBurst) + kExtraIteration;
+
+    sycl::device_ptr<TT> vector_ptr_device(l_device);
+
+    // Repeat matrix_count complete L matrix pipe reads
+    // for as many repetitions as needed
+    [[intel::loop_coalesce(2)]]
+    for (int repetition_index = 0; repetition_index < repetitions;
+                                                          repetition_index++) {
+      for (int matrix_index = 0; matrix_index < matrix_count; matrix_index++) {
+
+        for (int li = 0; li < kLoopIter; li++) {
+          TT bank[kNumElementsPerDDRBurst];
+
+          for (int k = 0; k < kNumElementsPerDDRBurst; k++) {
+            if(((li * kNumElementsPerDDRBurst) + k) < kLMatrixSize){
+              bank[k] = LMatrixPipe::read();
+            }
+          }
+
+          // Copy the L matrix result to DDR
+          if constexpr (kIncompleteBurst){
+            // Write a burst of kNumElementsPerDDRBurst elements to DDR
+            #pragma unroll
+            for (int k = 0; k < kNumElementsPerDDRBurst; k++) {
+              if(((li * kNumElementsPerDDRBurst) + k) < kLMatrixSize){
+                vector_ptr_device[(matrix_index * kLMatrixSize)
+                          + (li * kNumElementsPerDDRBurst) + k] = bank[k];
+              }
+            }
+          }
+          else{
+            // Write a burst of kNumElementsPerDDRBurst elements to DDR
+            #pragma unroll
+            for (int k = 0; k < kNumElementsPerDDRBurst; k++) {
+              vector_ptr_device[(matrix_index * kLMatrixSize)
+                          + (li * kNumElementsPerDDRBurst) + k] = bank[k];
+            }
+          }
+        }  // end of li
+      }  // end of matrix_index
+    }    // end of repetition_index
+  });
+
+  ddr_write_event.wait();
```

Comment:
Just add .wait() to end of q.single_task?

Reply:
I can't because I want to use this event later to compute the total execution time.
When using wait in the single_task I can't get the event (or I don't know how to do it)